ui/src/kapacitor/components/CodeData.js

```diff
@@ -0,0 +1,34 @@
+import React, {Component, PropTypes} from 'react'
+
+// needs to be React Component for click handler to work
+class CodeData extends Component {
```

Comment:
Good question. Changed it back to presentational and seems to work now. I may have tested this in the wrong order. Its parent does seem to need to be a `class` for templates to show up properly.